agent/api/container.go

```diff
@@ -135,7 +135,14 @@ type Container struct {
 	// metadata file
 	MetadataFileUpdated bool `json:"metadataFileUpdated"`
 
-	knownExitCode     *int
+	// KnownExitCodeUnsafe specifies the exit code for the container.
+	// It is exposed outside of the package so that it's marshalled/unmarshalled in
+	// the JSON body while saving the state.
+	// NOTE: Do not access KnownExitCodeUnsafe directly. Instead, use `GetKnownExitCode`
+	// and `SetKnownExitCode`.
+	KnownExitCodeUnsafe *int `json:"KnownExitCode"`
```

Comment:
You may also bump the state file version [here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/statemanager/state_manager.go#L60)